svo (curing skeleton, controllers, action system).xml

```diff
@@ -2237,34 +2237,130 @@ function svo.sendcuring(what)
+    len = len + #v
+  end
+  return len
+end
+
+sk.affqueueslow = {}
+sk.defqueueslow = {}
+sk.queuecount = sk.queuecount or 0
+function sk.dosendprios()
+    local sys = svo.sys
+    local function sendstuff(what, queue)
+      if queue == "normal" then
+        send(what,false)
+      elseif queue == "slow" then
+        if sys.sync == true then
+          sendAll("curingset switch slowcuring", what, false)
+        else
+          sendAll("curingset switch slowcuring", what, "curingset switch normal", false)
+        end
       end
-      what = "curing priority"..table.concat(sk.affqueue)
-      sk.sendqueuecmd(what)
-      sk.affqueue = {}
     end
-  elseif type == 'defence' then
-    if not table.contains(sk.defqueue, prio) then
-      table.insert(sk.defqueue, prio)
+    
+    if sk.sendpriotimer then killTimer( sk.sendpriotimer ) end
+  
+    if #sk.affqueue &gt; 0 then
+        local what = "curing priority"..table.concat(sk.affqueue)
+        if sk.queuecount &gt; 4 then
+          tempTimer(1,function()
+            sendstuff(what,"normal")
+            sk.affqueue = {}
+            sk.queuecount = 0
+          end)
+        else
+          sendstuff(what,"normal")
+          sk.affqueue = {}
+          sk.queuecount = sk.queuecount + 1
+        end
     end
-    if #sk.defqueue &gt;= 20 then
-      if #sk.sendqueue &gt;= 6 then
-        sk.dosendqueue()
-      end
-      what = "curing priority defence"..table.concat(sk.defqueue)
-      sk.sendqueuecmd(what)
-      sk.defqueue = {}
+    if #sk.defqueue &gt; 0 then
+        local what = "curing priority defence"..table.concat(sk.defqueue)
+        if sk.queuecount &gt; 4 then
+          tempTimer(1,function()
+          sendstuff(what,"normal")
+          sk.defqueue = {}
+          sk.queuecount = 0
+          end)
+        else
+          sendstuff(what,"normal")
+          sk.defqueue = {}
+          sk.queuecount = sk.queuecount + 1
+        end
+    end
+    if #sk.affqueueslow &gt; 0 then
+        local what = "curing priority"..table.concat(sk.affqueueslow)
+        if sk.queuecount &gt; 4 then
+          tempTimer(1,function()
+          sendstuff(what,"slow")
+          sk.affqueueslow = {}
+          sk.queuecount = 0
+          end)
+        else
+          sendstuff(what,"slow")
+          sk.affqueueslow = {}
+          sk.queuecount = sk.queuecount + 1
+        end
+    end
+    if #sk.defqueueslow &gt; 0 then
+        local what = "curing priority defence"..table.concat(sk.defqueueslow)
+        if sk.queuecount &gt; 4 then
+          tempTimer(1,function()
+          sendstuff(what,"slow")
+          sk.defqueueslow = {}
+          sk.queuecount = 0
+          end)
+        else
+          sendstuff(what,"slow")
+          sk.defqueueslow = {}
+          sk.queuecount = sk.queuecount + 1
+          display(sk.queuecount)
+        end
     end
+  
+  if queuetimer then killTimer(queuetimer) end
+  local queuetimer = tempTimer(1,function() sk.queuecount = 0 end)     
```

Comment:
How will this variable work? It's a local one that is gone at the end of the function.